mmv1/products/documentaiwarehouse/Location.yaml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
```

Comment:
It looks like `autogen_async: true` and `async` properties are needed as the initialize and get methods returns Operation based on API doc.

Terraform doc: https://googlecloudplatform.github.io/magic-modules/develop/resource/
Example: 
https://github.com/GoogleCloudPlatform/magic-modules/blob/d2ca0e13dfcd058cbb61d5d93a9599b191204c8a/mmv1/products/activedirectory/Peering.yaml#L29
https://github.com/GoogleCloudPlatform/magic-modules/blob/d2ca0e13dfcd058cbb61d5d93a9599b191204c8a/mmv1/products/activedirectory/Peering.yaml#L48C1-L48C20

Reply:
I don't find the get method for projects.locations.

Are you familiar with the document AI warehouse APIs?

Reply:
I have pushed the new changes now, please check

Reply:
Location is used as a resource to initialize an instance of DocAI Warehouse, which is a one-time thing for each project. Upon initialization, a long-running operation is created. Location only has a POST request right now, and once the POST request is successful, an instance of [Operation](https://cloud.google.com/document-warehouse/docs/reference/rest/v1/projects.locations.operations#resource:-operation) is given in the response body. As of now, I had used this operation-id to check for Location creation (ie, a DocAI Warehouse instance is provisioned or not). 

If Location is not initialized then we won't be able to use the service since this is an additional required step (after enabling the API)

Reply:
Thanks for the information. As the get method is not present, `immutable: true` and 
`skip_read: true` can be used to skip the read method.

https://github.com/GoogleCloudPlatform/magic-modules/blob/4d202133977a4d658f1f2bb94b6a4bda84c60d6f/mmv1/products/publicca/ExternalAccountKey.yaml#L21-L22

Reply:
Also, `exclude_import: true` can be used, as I don't see the use case to import the project location.